File: Spoj/MAJOR_Majority.py
from sys import stdin,stdout
from collections import *
# stdin = open("/home/shiva/Learning/1.txt","r")
# stdout = open("/home/shiva/Learning/2.txt","w")

def main():
	for a in range(int(stdin.readline().strip())):
		n = int(stdin.readline())
		arr = list(map(int,stdin.readline().strip().split()))
		pos = [0]*1001
		neg = [0]*1001
		for i in arr:
			if i<0:
				neg[-1*i]+=1
			else:
				pos[i]+=1

		val = 0; ii=0; vali = -1;
		for i in pos:
			if i>val:
				val = i
				vali = ii
		
			ii+=1
		if flag!=-1:
			stdout.write('YES '+str(flag)+'\n')
		else:
			stdout.write('NO\n')

main()

File: Spoj/MAJOR_Majority.cpp
#include <bits/stdc++.h>
#define vi vector<int>
#define pi pair<int,int>
#define vii vector<pi>
#define ll long long
using namespace std;
int main() {
#ifndef ONLINE_JUDGE
	freopen("/home/shiva/Learning/1.txt", "r", stdin);
	freopen("/home/shiva/Learning/2.txt", "w", stdout);
#endif
	ios::sync_with_stdio(false); cin.tie(0);
	int t; cin >> t; int x;
	while (t--) {
		int n; cin >> n;
		int pos[2005] = {0};
		for (int i = 0; i < n; i++) {
			cin >> x;
			pos[x + 1000]++;
		}
		int val = 0, vali = -1;
		for (int i = 0; i < 2005; i++)
			if (pos[i] >= val) {
				val = pos[i];
				vali = i;
			}
		if (val > n / 2) {
			cout << "YES " << vali - 1000 << endl;
		} else cout << "NO\n";
	}
}